website/content/en/docs/kubebuilder/quickstart.md

```diff
@@ -0,0 +1,763 @@
+
+> **// TODO:** Should we point to the client API doc after the integration?
+
+### Specify permissions and generate RBAC manifests
+
+The controller needs certain RBAC permissions to interact with the resources it manages. These are specified via [RBAC markers][rbac_markers] like the following:
+
+```Go
+// +kubebuilder:rbac:groups=cache.example.com,resources=memcacheds,verbs=get;list;watch;create;update;patch;delete
+// +kubebuilder:rbac:groups=cache.example.com,resources=memcacheds/status,verbs=get;update;patch
+// +kubebuilder:rbac:groups=apps,resources=deployments,verbs=get;list;watch;create;update;patch;delete
+// +kubebuilder:rbac:groups=core,resources=pods,verbs=get;list;
+
+func (r *MemcachedReconciler) Reconcile(req ctrl.Request) (ctrl.Result, error) {
+```
+
+The `ClusterRole` manifest at `config/rbac/role.yaml` is generated from the above markers via controller-gen with the following command:
+
+```sh
+$ make manifests
+```
+
+## Build and run the operator
+
+Before running the operator, the CRD must be registered with the Kubernetes apiserver:
+
+```sh
+$ make install
+```
+
+Once this is done, there are two ways to run the operator:
+
+- As Go program outside a cluster
+- As a Deployment inside a Kubernetes cluster
+
+### 1. Run locally outside the cluster
+
+To run the operator locally execute the following command:
+
+```sh
+$ make run ENABLE_WEBHOOKS=false
+```
+
+> **// TODO:** Add an advanced section or separate doc on writing defaulting/validating and conversion webhooks.
+
+### 2. Run as a Deployment inside the cluster
+
+#### Build and push the image
+
+Build the operator image and push it to a registry.
+
+Make sure to modify `quay.io/example/` in the example below to reference a container repository that
+you have access to. You can obtain an account for storing containers at
+repository sites such quay.io or hub.docker.com
+
+Build the image:
+
+```sh
+$ make docker-build IMG=quay.io/example/memcached-operator:v0.0.1
+```
+
+Push the image to a repository:
+
+```sh
+$ make docker-push IMG=quay.io/example/memcached-operator:v0.0.1
+```
+
+> **// TODO:** Mention how to use `buildah build` by just updating the Makefile?
+
+**Note**:
+The name and tag of the image (`IMG=<some-registry>/<project-name>:tag`) in both the commands can also be set in the Makefile. Modify the line which has `IMG ?= controller:latest` to set your desired default base image.
+
+#### Deploy the operator
+
+For this example we will run the operator in the `default` namespace which can be specified for all resources in `config/default/kustomization.yaml`:
+
+```YAML
+# Adds namespace to all resources.
+namespace: default
+...
+```
+
+Run the following to deploy the operator. This will also install the RBAC manifests from `config/rbac`.
+
+```sh
+$ make deploy IMG=quay.io/example/memcached-operator:v0.0.1
+```
+
+Verify that the memcached-operator is up and running:
+
+```sh
```

Comment:
Anything run from the shell with output looks better with a `console` tag:
```console
$ kubectl get deployment
NAME                                    READY   UP-TO-DATE   AVAILABLE   AGE
memcached-operator-controller-manager   1/1     1            1           8m
```
vs
```sh
$ kubectl get deployment
NAME                                    READY   UP-TO-DATE   AVAILABLE   AGE
memcached-operator-controller-manager   1/1     1            1           8m
```